app/controllers/appointment_controller.rb

```diff
@@ -0,0 +1,20 @@
+require 'date'
+class AppointmentController < ApplicationController
+    def create_appointment
+        doa = DateTime.parse("#{params[:date]}T#{params[:time]}")
+        @appointment = Appointment.create(
+          date_of_appointment: doa,
+          no_of_session: params[:no_of_session],
+          doctor_id: params[:doctor_id],
+          patient_id: params[:patient_id],
+          status: params[:status],
+        )
+
+        respond_to do |format|
+            format.html
+            format.json do
+                render json(accepted: true)
+            end
+        end
+    end
```

Comment:
You should have a private method that specify the kind of params you're expecting like 
```ruby
private
def appointment_params
  params.require(:time, :date, :doctor_id)
end
```

then instead of calling params in your create method you'll call

```ruby
  appointment_params[:date]
```

Reply:
So if the params are not sent it just show an error, which you can test in the other context